Update core to latest

asuresh4 opened the pull request:
Run `make update-otel`

asuresh4 committed:
Update core to latest

asuresh4 commented on go.mod:
```diff
@@ -76,8 +71,6 @@ replace github.com/open-telemetry/opentelemetry-collector-contrib/exporter/azure
 
 replace github.com/open-telemetry/opentelemetry-collector-contrib/exporter/carbonexporter => ./exporter/carbonexporter
 
-replace github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter => ./exporter/datadogexporter
```
Removed this line since the exporter is not yet enabled.

tigrannajaryan commented:
@asuresh4 the loadtest is consistently failing. We may have a performance regression. Can you have a look?

asuresh4 commented:
The recent failures are from OC traces. Doesn't look like we've had changes since last release though - [exporter](https://github.com/open-telemetry/opentelemetry-collector/commits/master/exporter/opencensusexporter) and [receiver] (https://github.com/open-telemetry/opentelemetry-collector/commits/master/receiver/opencensusreceiver). Can look into it more but since the memory usage is only over 1-2 MB, could it just be bad luck with CI?

tigrannajaryan commented:
> Can look into it more but since the memory usage is only over 1-2 MB, could it just be bad luck with CI?

@asuresh4 I re-run a few times on CI and it failed all the time. Doesn't look like bad luck. Also it can be due to dependency changes. Can you please check older builds and see if there is any significant increase? It could be that we were very close to the limit and there is a tiny change which causes failures, in which case it is fine.

I want to make sure we do not see dramatic memory usage increase.

asuresh4 commented:
> > Can look into it more but since the memory usage is only over 1-2 MB, could it just be bad luck with CI?
> 
> @asuresh4 I re-run a few times on CI and it failed all the time. Doesn't look like bad luck. Also it can be due to dependency changes. Can you please check older builds and see if there is any significant increase? It could be that we were very close to the limit and there is a tiny change which causes failures, in which case it is fine.
> 
> I want to make sure we do not see dramatic memory usage increase.

I think you're right @tigrannajaryan, there seems to be a consistent increase across all successful tests. I took sample of 5 from recent successful builds, comparing those to the latest builds that failed from this PR, the average RAM usage seems to have gone up slightly (4-6MB) across tests.

Old test results.

```
Test                                    |Result|Duration|CPU Avg%|CPU Max%|RAM Avg MiB|RAM Max MiB|Sent Items|Received Items|
----------------------------------------|------|-------:|-------:|-------:|----------:|----------:|---------:|-------------:|
Metric10kDPS/OpenCensus                 |PASS  |     15s|    52.4|    55.3|         51|         61|   1049300|       1049300|
Metric10kDPS/Carbon                     |PASS  |     17s|   210.7|   214.9|         54|         65|    562800|        562800|
Metric10kDPS/SignalFx                   |PASS  |     15s|   104.4|   105.8|         57|         64|   1019200|       1019200|
Trace10kSPS/OpenCensus                  |PASS  |     15s|    22.9|    24.5|         55|         66|    149900|        149900|
Trace10kSPS/OTLP                        |PASS  |     15s|     9.5|    11.0|         47|         57|    149900|        149900|
Trace10kSPS/SAPM                        |PASS  |     15s|    22.6|    24.0|         59|         72|    149900|        149900|

Test                                    |Result|Duration|CPU Avg%|CPU Max%|RAM Avg MiB|RAM Max MiB|Sent Items|Received Items|
----------------------------------------|------|-------:|-------:|-------:|----------:|----------:|---------:|-------------:|
Metric10kDPS/OpenCensus                 |PASS  |     15s|    50.1|    53.3|         49|         59|   1049300|       1049300|
Metric10kDPS/Carbon                     |PASS  |     17s|   221.4|   224.7|         53|         63|    620900|        620900|
Metric10kDPS/SignalFx                   |PASS  |     15s|   106.5|   108.3|         59|         64|   1050000|       1050000|
Trace10kSPS/OpenCensus                  |PASS  |     15s|    22.5|    23.1|         54|         66|    149900|        149900|
Trace10kSPS/OTLP                        |PASS  |     15s|     9.7|    10.6|         49|         60|    149900|        149900|
Trace10kSPS/SAPM                        |PASS  |     15s|    21.8|    23.1|         60|         73|    149900|        149900|

Test                                    |Result|Duration|CPU Avg%|CPU Max%|RAM Avg MiB|RAM Max MiB|Sent Items|Received Items|
----------------------------------------|------|-------:|-------:|-------:|----------:|----------:|---------:|-------------:|
Metric10kDPS/OpenCensus                 |PASS  |     15s|    53.1|    54.0|         51|         61|   1035300|       1035300|
Metric10kDPS/Carbon                     |PASS  |     18s|    83.4|    94.2|         54|         63|    216300|        216300|
Metric10kDPS/SignalFx                   |PASS  |     15s|    58.2|    62.0|         53|         65|    534100|        534100|
Trace10kSPS/OpenCensus                  |PASS  |     15s|    22.8|    24.9|         55|         66|    149800|        149800|
Trace10kSPS/OTLP                        |PASS  |     15s|     9.9|    10.9|         47|         56|    149900|        149900|
Trace10kSPS/SAPM                        |PASS  |     15s|    20.9|    22.3|         61|         74|    148800|        148800|

Test                                    |Result|Duration|CPU Avg%|CPU Max%|RAM Avg MiB|RAM Max MiB|Sent Items|Received Items|
----------------------------------------|------|-------:|-------:|-------:|----------:|----------:|---------:|-------------:|
Metric10kDPS/OpenCensus                 |PASS  |     15s|    41.4|    44.6|         50|         60|   1049300|       1049300|
Metric10kDPS/Carbon                     |PASS  |     16s|   196.2|   201.4|         53|         63|    695800|        695800|
Metric10kDPS/SignalFx                   |PASS  |     15s|    98.5|   101.7|         58|         62|   1050000|       1050000|
Trace10kSPS/OpenCensus                  |PASS  |     15s|    21.5|    22.3|         54|         65|    149900|        149900|
Trace10kSPS/OTLP                        |PASS  |     15s|     8.3|     9.3|         48|         58|    149900|        149900|
Trace10kSPS/SAPM                        |PASS  |     15s|    20.3|    21.5|         58|         71|    150000|        150000|

Test                                    |Result|Duration|CPU Avg%|CPU Max%|RAM Avg MiB|RAM Max MiB|Sent Items|Received Items|
----------------------------------------|------|-------:|-------:|-------:|----------:|----------:|---------:|-------------:|
Metric10kDPS/OpenCensus                 |PASS  |     15s|    53.9|    57.6|         51|         60|   1049300|       1049300|
Metric10kDPS/Carbon                     |PASS  |     16s|   212.2|   215.4|         52|         62|    581000|        581000|
Metric10kDPS/SignalFx                   |PASS  |     15s|   112.2|   114.7|         53|         63|   1036700|       1036700|
Trace10kSPS/OpenCensus                  |PASS  |     15s|    24.5|    26.0|         54|         65|    149900|        149900|
Trace10kSPS/OTLP                        |PASS  |     15s|     9.9|    11.2|         46|         56|    150000|        150000|
Trace10kSPS/SAPM                        |PASS  |     15s|    23.2|    25.3|         59|         71|    149900|        149900|
```



Latest tests that failed.
```
Test                                    |Result|Duration|CPU Avg%|CPU Max%|RAM Avg MiB|RAM Max MiB|Sent Items|Received Items|
----------------------------------------|------|-------:|-------:|-------:|----------:|----------:|---------:|-------------:|
Metric10kDPS/OpenCensus                 |PASS  |     15s|    49.7|    52.7|         54|         66|   1049300|       1049300|
Metric10kDPS/Carbon                     |PASS  |     17s|   215.4|   219.4|         59|         70|    586600|        586600|
Metric10kDPS/SignalFx                   |PASS  |     15s|   106.4|   107.0|         59|         70|   1040900|       1040900|
Trace10kSPS/OpenCensus                  |FAIL  |      6s|    23.2|    23.8|         50|         72|     59300|         58500|RAM consumption is 72 MiB, max expected is 70 MiB
Trace10kSPS/OTLP                        |PASS  |     15s|     9.0|    10.3|         53|         64|    149900|        149900|
Trace10kSPS/SAPM                        |PASS  |     15s|    21.0|    22.0|         64|         77|    149900|        149900|

Test                                    |Result|Duration|CPU Avg%|CPU Max%|RAM Avg MiB|RAM Max MiB|Sent Items|Received Items|
----------------------------------------|------|-------:|-------:|-------:|----------:|----------:|---------:|-------------:|
Metric10kDPS/OpenCensus                 |PASS  |     15s|    56.6|    58.0|         55|         66|   1049300|       1049300|
Metric10kDPS/Carbon                     |PASS  |     17s|   208.8|   213.7|         56|         68|    521500|        521500|
Metric10kDPS/SignalFx                   |PASS  |     15s|   108.3|   110.8|         62|         71|    981400|        981400|
Trace10kSPS/OpenCensus                  |FAIL  |      6s|    25.6|    26.3|         48|         71|     59300|         59000|RAM consumption is 71 MiB, max expected is 70 MiB
Trace10kSPS/OTLP                        |PASS  |     15s|    11.1|    12.9|         55|         67|    149900|        149900|
Trace10kSPS/SAPM                        |PASS  |     15s|    24.8|    26.5|         66|         79|    149900|        149900|

Test                                    |Result|Duration|CPU Avg%|CPU Max%|RAM Avg MiB|RAM Max MiB|Sent Items|Received Items|
----------------------------------------|------|-------:|-------:|-------:|----------:|----------:|---------:|-------------:|
Metric10kDPS/OpenCensus                 |PASS  |     15s|    49.6|    52.0|         56|         66|   1049300|       1049300|
Metric10kDPS/Carbon                     |PASS  |     17s|   213.8|   217.7|         57|         68|    578200|        578200|
Metric10kDPS/SignalFx                   |PASS  |     15s|   103.4|   106.7|         58|         70|   1021300|       1021300|
Trace10kSPS/OpenCensus                  |FAIL  |      9s|    22.2|    23.0|         53|         71|     89200|         89000|RAM consumption is 71 MiB, max expected is 70 MiB
Trace10kSPS/OTLP                        |PASS  |     15s|     9.6|    11.0|         53|         64|    149900|        149900|
Trace10kSPS/SAPM                        |FAIL  |     15s|    22.9|    23.3|         67|         81|    150000|        149400|RAM consumption is 81 MiB, max expected is 80 MiB
```

tigrannajaryan commented:
@asuresh4 thanks for looking. Can you also check the tests results in the core repo? We have the same tests there, unclear they didn't fail there. Could be because the limits are different or could be that in contrib they are actually slower.

tigrannajaryan commented:
Good thing that I don't see CPU usage regressions, but RAM increase is unclear.

asuresh4 commented:
RAM is usage for tests in core are <70MB (in the upper 60s for recent builds). However, the expected max. RAM is set to 82MB (https://github.com/open-telemetry/opentelemetry-collector/blob/master/testbed/tests/trace_test.go#L64). @tigrannajaryan - do you think contrib should have the same? But I don't think that still wouldn't explain why the usage is different in contrib vs core.

tigrannajaryan commented:
@asuresh4 supposedly contrib should use the same memory as core for the same config. Can you check if we saw an increase of memory usage in the core since contrib was last updated? Perhaps it is below the limit in the core but there is still an increase visible?

asuresh4 commented:
@tigrannajaryan - I was able to narrow to two subsequent builds where the increase is first visible in core

1) https://app.circleci.com/pipelines/github/open-telemetry/opentelemetry-collector/3730/workflows/4b886032-dc59-4689-a5bc-4bd8e0a95a4b/jobs/40211 - triggered by https://github.com/open-telemetry/opentelemetry-collector/pull/1863
2) https://app.circleci.com/pipelines/github/open-telemetry/opentelemetry-collector/3743/workflows/7e9293c0-cb69-4f44-a9d2-9fe21d6ce97a/jobs/40419 - triggered by https://github.com/open-telemetry/opentelemetry-collector/pull/1877

It's between these two commits in core, where the avg. RAM usage seems to have gone up by ~20MB. It unclear why though since the change from https://github.com/open-telemetry/opentelemetry-collector/pull/1877 seems unrelated.

Pushing a commit to bump expected RAM max to match core.

tigrannajaryan commented:
OK, doesn't seem like we can easily figure out why this happened. Most likely triggered by a change in a dependency. I don't want to fight it though. CPU is not increased, which is much more important. Whatever the dependency change is I don't want to block that and stay on some old version. It is not worth it. We can return to this topic if we ever need to look carefully at total memory usage.

asuresh4 committed:
Merge remote-tracking branch 'upstream/master' into update-core-version
